test/retorts.py

```diff
@@ -29,51 +29,14 @@
 )
 
 
-class ResponsesHelper:
-    """
-    Work around the lack of reentrance in the `responses` library. Both `moto`
-    and `responses` suffer from an inherent design flaw: they use
-    unittest.mock.patch which is global but their decorators and context
-    managers pretend to be reentrant. It is essentially impossible to nest a
-    RequestMock CM inside another one, or in a method decorated with
-    @responses.activate or one of the moto decorators. Furthermore, one can't
-    combine @responses.activate with any of the moto decorators. Use this
-    method in tests as follows:
-
-    >>> import moto
-    >>> import requests
-    >>> @moto.mock_sts
-    ... def test_foo():
-    ...     with ResponsesHelper() as helper:
-    ...         helper.add(responses.Response(method=responses.GET,
-    ...                                       url='http://foo.bar/blah',
-    ...                                       body='Duh!'))
-    ...         helper.add_passthru('http://localhost:12345/')
-    ...         assert requests.get('http://foo.bar/blah').content == b'Duh!'
-    >>> test_foo()
-
-    >>> with ResponsesHelper() as helper: #doctest: +ELLIPSIS
-    ...     pass
-    Traceback (most recent call last):
-    ...
-    AssertionError: This helper only works with `responses` already active. ...
-
-    In other words, whenever you would call the global responses.add() or
-    responses.add_passthru() functions, call the helper's method of the same
-    name instead. The helper will remove the mock response upon exit and
-    restore the set pf pass-throughs to their original value, essentially
-    undoiing all .add() and .add_passthru() invocations.
-
-    Remember that you do not need @responses.activate if one of the moto
-    decorators is present since they already activate responses.
-    """
-
-    def __init__(self, request_mock: responses.RequestsMock = None) -> None:
-        super().__init__()
+class AuthResponseHelper:
```

Comment:
Seems that some part of the old docstring should be preserved here. Maybe just the last two sentences.

Reply:
Check this out, let me know if additional changes should be made.